sidecar.go

```diff
@@ -45,27 +50,55 @@ const (
 	TelegrafLimitsCPU = "telegraf.influxdata.com/limits-cpu"
 	// TelegrafLimitsMemory allows specifying custom memory resource limits
 	TelegrafLimitsMemory = "telegraf.influxdata.com/limits-memory"
-	telegrafSecretPrefix = "telegraf-config"
+	telegrafSecretInfix  = "config"
 )
 
 type sidecarHandler struct {
+	ClassDataHandler            *classDataHandler
 	Logger                      logr.Logger
+	TelegrafDefaultClass        string
 	TelegrafImage               string
 	EnableDefaultInternalPlugin bool
 	RequestsCPU                 string
 	RequestsMemory              string
 	LimitsCPU                   string
 	LimitsMemory                string
+	EnableIstioInjection        bool
+	IstioOutputClass            string
+}
+
+type sidecarHandlerResponse struct {
+	// list of secrets to create alongside with the changes
+	secrets []*corev1.Secret
 }
 
 // This function check if the pod have the correct annotations, otherwise the controller will skip this pod entirely
 func (h *sidecarHandler) skip(pod *corev1.Pod) bool {
+	return !h.shouldAddTelegrafSidecar(pod) && !h.shouldAddIstioTelegrafSidecar(pod)
```

Comment:
```suggestion
	return h.shouldAddTelegrafSidecar(pod) || h.shouldAddIstioTelegrafSidecar(pod)
```

The logic with the `sidecar.istio.io/status` is that this webhook can be called multiple times during pod creation:
- before and after istio own webhook
- only after istio webook

we would need to handle adding only 1 sidecar at the time. ignoring the other one at any given time

Reply:
I'm thinking that we can even skip this global `skip` function and only using both function. We should also check the precence of a container called `telegraf` to check if this webhook as already run. same goes for the istio sidecar. if the annotation is present but he container `telegraf-istio` is already present, do nothing

Reply:
```
-	return !h.shouldAddTelegrafSidecar(pod) && !h.shouldAddIstioTelegrafSidecar(pod)
+	return h.shouldAddTelegrafSidecar(pod) || h.shouldAddIstioTelegrafSidecar(pod)
```
Logical error here. That would still require negation - `!X && !Y == !(X || Y)`.

Or, it could be done by changing the name `skip` to `shouldAddSidecars` and then changing `if a.SidecarHandler.skip(pod) {` to `if !a.SidecarHandler.skip(pod) {`. But I would leave the `skip` function as is (i.e. not revert its return value) because we already have tests for that.

As for removing `skip` entirely - I like that we never add any changes without the annotation - for example not doing the `pod.SetName()` when `generateName` was specified.

If you are strong on removing `skip`, I can do it, but I liked the idea that we do not modify pods that have no related annotations.

Reply:
you are right on the logical error, my bad. We can leave `skip` just making sure to include the container name existence in the function I guess?